x-pack/plugins/lens/public/datasources/text_based/utils.ts

```diff
@@ -64,24 +65,50 @@ export async function getStateFromAggregateQuery(
   query: AggregateQuery,
   dataViews: DataViewsPublicPluginStart,
   data: DataPublicPluginStart,
-  expressions: ExpressionsStart
+  expressions: ExpressionsStart,
+  frameDataViews?: DataViewsState
```

Comment:
Nit: Will `frameDataViews` only contain ad-hoc data views? Maybe we could just call them that (`adHocDataViews`) since that seems to be the term we all use?

Reply:
It contains everything, it actually stores the `frame.dataViews`. Lens stores dataviews in many places, we should clean this up in the future but on my tests frame.dataViews is the most correct one. I think the name is ok for now